java/src/main/java/com/logicalclocks/hsfs/engine/StreamFeatureGroupEngine.java

```diff
@@ -65,30 +66,100 @@ public <S> StreamFeatureGroup save(StreamFeatureGroup featureGroup, S dataset, L
+        SaveMode.APPEND,  partitionKeys, hudiPrecombineKey, writeOptions, sparkJobConfiguration);
+
+    return featureGroup;
+  }
+
+  @SneakyThrows
+  public <S> Object insertStream(StreamFeatureGroup streamFeatureGroup, S featureData, String queryName,
+                                 String outputMode, boolean awaitTermination, Long timeout,  String checkpointLocation,
+                                 List<String> partitionKeys, String hudiPrecombineKey, Map<String, String>
+                                       writeOptions, JobConfiguration jobConfiguration) {
+
+    if (streamFeatureGroup.getValidationType() != ValidationType.NONE) {
+      LOGGER.info("ValidationWarning: Stream ingestion for feature group `" + streamFeatureGroup.getName()
+                    + "`, with version `" + streamFeatureGroup.getVersion() + "` will not perform validation.");
+    }
+
+    if (writeOptions == null) {
+      writeOptions = new HashMap<>();
+    }
+
+    if (streamFeatureGroup.getId() == null) {
+      streamFeatureGroup = saveFeatureGroupMetaData(streamFeatureGroup, partitionKeys, hudiPrecombineKey, writeOptions,
+          jobConfiguration, featureData, true);
+    }
+
+    return SparkEngine.getInstance().writeStreamDataframe(streamFeatureGroup,
+      utils.sanitizeFeatureNames(featureData), queryName, outputMode, awaitTermination, timeout, checkpointLocation,
+      utils.getKafkaConfig(streamFeatureGroup, writeOptions));
+  }
+
+  public <S> void insert(StreamFeatureGroup streamFeatureGroup, S featureData, HudiOperationType operation,
+                         SaveMode saveMode, List<String> partitionKeys, String hudiPrecombineKey,
+                         Map<String, String> writeOptions, JobConfiguration jobConfiguration)
+      throws FeatureStoreException, IOException, ParseException {
+
+    Integer validationId = null;
+    if (streamFeatureGroup.getValidationType() != ValidationType.NONE) {
+      FeatureGroupValidation validation = streamFeatureGroup.validate(featureData, true);
+      if (validation != null) {
+        validationId = validation.getValidationId();
+      }
+    }
+
+    if (streamFeatureGroup.getId() == null) {
+      streamFeatureGroup = saveFeatureGroupMetaData(streamFeatureGroup, partitionKeys, hudiPrecombineKey, writeOptions,
+          jobConfiguration, featureData, false);
+    }
+
+    if (saveMode == SaveMode.OVERWRITE) {
+      // If we set overwrite, then the directory will be removed and with it all the metadata
+      // related to the feature group will be lost. We need to keep them.
+      // So we call Hopsworks to manage to truncate the table and re-create the metadata
+      // After that it's going to be just a normal append
+      featureGroupApi.deleteContent(streamFeatureGroup);
+    }
+
+    SparkEngine.getInstance().writeOnlineDataframe(streamFeatureGroup, featureData,
+        streamFeatureGroup.getOnlineTopicName(), utils.getKafkaConfig(streamFeatureGroup, writeOptions));
+  }
```

Comment:
This method has the same issue as the one from the `FeatureGroupEngine`. Additionally it seems that the result of the validation (validationId) is ignored - probably on purpose since the `insertStream` method above throws an exception, but then we should be consistent.